File: backend/app/schemas/job.py
"""ProcessingJob Pydantic schemas."""

import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.models.enums import JobStatus


class JobResponse(BaseModel):
    id: uuid.UUID
    meeting_id: uuid.UUID
    status: JobStatus
    stage: str | None
    attempt_count: int
    error_message: str | None
    created_at: datetime
    started_at: datetime | None
    completed_at: datetime | None

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/job.ts
export type JobStatusType = 'queued' | 'processing' | 'completed' | 'failed' | 'cancelled';

export interface Job {
  id: string;
  meeting_id: string;
  status: JobStatusType;
  stage: string | null;
  attempt_count: number;
  error_message: string | null;
  created_at: string;
  started_at: string | null;
  completed_at: string | null;
}
